flink-tests/src/test/java/org/apache/flink/test/streaming/runtime/PartitionerITCase.java

```diff
@@ -228,9 +229,29 @@ private static void verifyRebalancePartitioning(List<Tuple2<Integer, String>> re
 				new Tuple2<Integer, String>(2, "c"),
 				new Tuple2<Integer, String>(0, "a"));
 
-		assertEquals(
-				new HashSet<Tuple2<Integer, String>>(expected),
-				new HashSet<Tuple2<Integer, String>>(rebalancePartitionResult));
+		List<Tuple2<Integer, String>> expected1 = Arrays.asList(
+				new Tuple2<Integer, String>(1, "a"),
+				new Tuple2<Integer, String>(2, "b"),
+				new Tuple2<Integer, String>(0, "b"),
+				new Tuple2<Integer, String>(1, "a"),
+				new Tuple2<Integer, String>(2, "a"),
+				new Tuple2<Integer, String>(0, "c"),
+				new Tuple2<Integer, String>(1, "a"));
+
+		List<Tuple2<Integer, String>> expected2 = Arrays.asList(
+				new Tuple2<Integer, String>(2, "a"),
+				new Tuple2<Integer, String>(0, "b"),
+				new Tuple2<Integer, String>(1, "b"),
+				new Tuple2<Integer, String>(2, "a"),
+				new Tuple2<Integer, String>(0, "a"),
+				new Tuple2<Integer, String>(1, "c"),
+				new Tuple2<Integer, String>(2, "a"));
+
+		assertTrue(
```

Comment:
to be more robust, shall this assert that exactly one case match and the other two unmatch?